doc/specs/#7335 - Console Allocation Policy.md

```diff
@@ -141,25 +187,42 @@ This should have no impact on performance, power or efficiency.
 
 I am **not** proposing a change in how shells determine whether to wait for an application before returning to a prompt.
 This means that a console subsystem application that intends to primarily present a UI but occasionally print text to a
-console (therefore choosing the **inheritOnly** allocation policy) will cause the shell to "hang" and wait for it to
+console (therefore choosing the **detached** allocation policy) will cause the shell to "hang" and wait for it to
 exit.
 
+The decision to pause/wait is made entirely in the calling shell, and the console subsystem cannot influence that
+decision.
+
 Because the vast majority of shells on Windows "hang" by calling `WaitFor...Object` with a HANDLE to the spawned
 process, an application that wants to be a "hybrid" CUI/GUI application will be forced to spawn a separate process to
 detach from the shell and then terminate its main process.
 
 This is very similar to the forking model seen in many POSIX-compliant operating systems.
 
-### Launching interactively from Explorer
+### Launching interactively from Explorer, Task Scheduler, etc.
 
-Applications like PowerShell may wish to retain "automatic" console allocation, and **inheritOnly** would be unsuitable
-for them. If PowerShell becomes **inheritOnly**, double-clicking `pwsh.exe` will no longer spawn a console. This would
-be a critical failure of the console subsystem.
+Applications like PowerShell may wish to retain automatic console allocation, and **detached** would be unsuitable for
+them. However, they _can_ use **hidden**. There is a problem, though: if PowerShell becomes **hidden**, double-clicking
+`pwsh.exe` will no longer spawn a console. This would almost certainly break PowerShell for all users.
 
 At the same time, PowerShell wants `-WindowStyle Hidden` to suppress the console _before it's created_.
 
-PowerShell, and any other shell that wishes to maintain interactive launch from the graphical shell, will need to call
-[`AllocConsole`] if it determines that it wants to launch in interactive mode.
+PowerShell, and any other shell that wishes to maintain interactive launch from the graphical shell, can start in
+**hidden** mode and then display its window as necessary. Therefore:
+
+* PowerShell will set `<consoleWindowPolicy>hidden</consoleWindowPolicy>`
+* On startup, it will process its commandline arguments.
+* If `-WindowStyle Hidden` is present, it will do nothing.

```

Comment:
side-bar: `-WindowStyle Hidden` just plain won't work with defterm, will it now?